Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ThreadSafeMongoMobile.swift

```diff
@@ -0,0 +1,130 @@
+import MongoSwift
+import StitchCoreSDK
+import StitchCoreLocalMongoDBService
+import Foundation
+
+class SyncMongoClient {
```

Comment:
[nit] I think a more appropriate prefix for these types would be "ThreadSafe" instead of "Sync", since that's actually what they are. They don't necessarily have anything to do with "Sync"

Reply:
And that way, once the Swift driver makes MongoClient thread-safe, it'll be very clear that we can get rid of this

Reply:
Technically, when I used Sync here, I meant `synchronised`. Which is what it's doing, but I take your point. It's also more than just "ThreadSafe" though. So I'm not sure what to name it.

Reply:
What about atomic?